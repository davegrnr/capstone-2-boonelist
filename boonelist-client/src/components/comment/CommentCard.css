.CommentCard {
	max-width: 600px;
	word-wrap: break-word;
	vertical-align: middle;
}

.remove-comment-btn {
	align-items: center;
}

#comment-border {
	border: 1.5px groove lightslategray;
	border-radius: 20px;
	margin-bottom: 3px;
	margin-top: 10px;
}
