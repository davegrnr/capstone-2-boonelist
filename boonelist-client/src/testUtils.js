import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  phone: "111-222-3333",
};

const UserProvider =
    ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
