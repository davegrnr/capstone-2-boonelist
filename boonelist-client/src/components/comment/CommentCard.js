import React, {useContext} from "react"
import { useParams } from "react-router-dom";
import BoonelistApi from "../../api/api";
import UserContext from "../../auth/UserContext";
import './CommentCard.css'

function CommentCard({ commentId, route, commentText, postedBy, createdAt, getSale, getService }){
    const currentUser = useContext(UserContext).currentUser
    const {id} = useParams()

    async function handleRemoveComment(){
        try{
            await BoonelistApi.removeComment(route, id, commentId)
            if(route === 'services'){
                getService(id)
            } else {
                getSale(id)
            }
        } catch(err){
            return (err)
        }
    }

    function commentNotOwnedDisplay() {
        return (
            <div className="CommentCard">
                <div id="comment-border">
                    <div className="card-body">
                        <h4><b>{postedBy}</b></h4>
                        <h6>{createdAt}</h6>
                        <p>{commentText}</p>
                    </div>
                </div>
            </div>
        )
    }

    function commentOwnedDisplay() {
        return (
            <div className="CommentCard">
                <div id="comment-border">
                    <div className="card-body">
                        <h4><b>{postedBy}</b></h4>
                        <h6>{createdAt}</h6>
                        <p>{commentText}</p>
                        <button className="btn btn-danger btn-sm remove-comment-btn"
                                onClick={handleRemoveComment}
                        >Remove</button>
                    </div>

                </div>
            </div>
        )
    }

    return (
        <div>
            {currentUser.username === postedBy ? commentOwnedDisplay() : commentNotOwnedDisplay()}
        </div>
    )

}

export default CommentCard;