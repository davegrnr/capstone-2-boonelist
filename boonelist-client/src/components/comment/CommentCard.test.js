import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import CommentCard from "./CommentCard";
import { UserProvider } from "../../testUtils";
import { it } from "@jest/globals";

it("matches a snapshot", function () {
    let item = { commentText: "text", postedBy: "user" };
    const { asFragment } = render(
        <MemoryRouter initialEntries={[`sales/1`]}>
            <UserProvider>
            <CommentCard item={item}/>
            </UserProvider>
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });