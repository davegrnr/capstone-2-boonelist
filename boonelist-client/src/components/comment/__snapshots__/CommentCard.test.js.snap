// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="CommentCard"
    >
      <div
        id="comment-border"
      >
        <div
          class="card-body"
        >
          <h4>
            <b />
          </h4>
          <h6 />
          <p />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
